sessionctx/variable/session.go

```diff
@@ -385,6 +385,9 @@ type SessionVars struct {
 
 	// Killed is a flag to indicate that this query is killed.
 	Killed uint32
+
+	// ConnectionInfo indicates current connection info used by current session, only be lazy assigned by plugin.
+	ConnectionInfo *ConnectionInfo
```

Comment:
How about saving it to `clientConn`?
It's not used in `sessionVars` anyway.

Reply:
@tiancaiamao clientConn is not easy to obtain in the later process phase(e.g. write audit log after a query finished - - ), it will be used in audit plugin's OnGeneralEvent